### DISCUSSION TITLE
New node - Display in center of screen

### DISCUSSION THREAD

**Discussion by brenolapreza** [xyflow/xyflow#2207] [Q&A] [answered]
Hi,

I have a certain problem, in the question of the positions of the nodes. I'm using custom nodes, when I click on the "+" button I add the node to the flow. However, I want it to be added to the center of the screen. How can I do this?

**Comment by moklick** [comment, marked as answer]
You can use the [`project` function](https://reactflow.dev/docs/api/react-flow-instance/#project) to convert screen coordinates into React Flow coordinates. If you would have a fullscreen pane, it would look like:

```js
const { x, y } = project({ x: window.innerWidth / 2, y: window.innerHeight / 2 });
```

**Reply by brenolapreza** [reply]
Thanks for the answer,

That way the new node would be centered on the screen at a fixed point, if I scroll the panel and add a new node, it wouldn't show up in the center after scrolling.

No matter where the user is on the panel, I want the new node to appear in the center. Any suggestion?

Thanks!

**Reply by bcakmakoglu** [reply]
That's either not accurate or I am misunderstanding you.

Implementing the projection Moritz suggested, it doesn't matter where you scroll, the new node will be added at the center wherever you currently are and not at a "fixed" position on the pane like x: 50, y: 50.

[Here's a sandbox showing the use of `project`.](https://codesandbox.io/s/sandpack-project-forked-z8b7nb?file=/App.js)

**Reply by moklick** [reply]
As @bcakmakoglu pointed out, the `project` function would give you different `x`, and `y` values for `{ x: window.innerWidth / 2, y: window.innerHeight / 2 }` when you zoom or pan.

**Reply by brenolapreza** [reply]
Sorry, I'm just getting started.

That's exactly what I need, I'm using v9, I don't think it works the same way. I will try to update to v10.

Thank you so much guys.
